contracts/voting-verifier/src/execute.rs

```diff
@@ -92,54 +80,46 @@ pub fn verify_messages(
 
     let config = CONFIG.load(deps.storage)?;
 
-    let messages = messages
-        .into_iter()
-        .map(|message| {
-            is_message_verified(deps.as_ref(), &message).map(|verified| {
-                if verified {
-                    Verified(message)
-                } else {
-                    Pending(message)
-                }
-            })
-        })
-        .collect::<Result<Vec<VerificationStatus>, ContractError>>()?;
-
     let response = Response::new().set_data(to_binary(&VerifyMessagesResponse {
-        verification_statuses: messages
-            .iter()
-            .map(|status| match status {
-                Verified(message) => (message.cc_id.clone(), true),
-                Pending(message) => (message.cc_id.clone(), false),
-            })
-            .collect(),
+        verification_statuses: is_verified(deps.as_ref(), &messages)?,
     })?);
 
-    let pending_messages: Vec<Message> = messages
+    let messages = messages
         .into_iter()
-        .filter_map(|status| match status {
-            Pending(message) => Some(message),
-            Verified(_) => None,
+        .map(|message| verification_status(deps.as_ref(), &message).map(|status| (status, message)))
+        .collect::<Result<Vec<(VerificationStatus, Message)>, ContractError>>()?;
+
```

Comment:
```suggestion
        .collect::<Result<Vec<_>, _>>()?;
```
Don't provide what the compiler can infer.